src/arch/wasm/CodeGen.zig

```diff
@@ -0,0 +1,1710 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const ArrayList = std.ArrayList;
+const assert = std.debug.assert;
+const testing = std.testing;
+const leb = std.leb;
+const mem = std.mem;
+const wasm = std.wasm;
+
+const Module = @import("../../Module.zig");
+const Decl = Module.Decl;
+const Type = @import("../../type.zig").Type;
+const Value = @import("../../value.zig").Value;
+const Compilation = @import("../../Compilation.zig");
+const LazySrcLoc = Module.LazySrcLoc;
+const link = @import("../../link.zig");
+const TypedValue = @import("../../TypedValue.zig");
+const Air = @import("../../Air.zig");
+const Liveness = @import("../../Liveness.zig");
+const Mir = @import("Mir.zig");
+const Emit = @import("Emit.zig");
+
+/// Wasm Value, created when generating an instruction
+const WValue = union(enum) {
```

Comment:
A minor nit but do we save much by abbreviating `WasmValue` to `WValue`?

Reply:
Back then I abbreviated it to keep it consistent with `MCValue` and `CValue`. Perhaps now is a good time to change it to `WasmValue`. I don't particularly mind either way.

Reply:
I'm fine either way. Just wanted to point out that we don't really save that much after all :-)